### PR TITLE
Fix sending multiple segment messages

### PR DESCRIPTION
**Pull request by affederaffe**
When sending non-single-segmented buffers, only the first segment was incorrectly sent over and over again.

### CODE REVIEW

File: src/Tmds.DBus.Protocol/SignatureReader.cs [hunk 1]
```diff
@@ -1,8 +1,6 @@
-[assembly: InternalsVisibleTo("dotnet-dbus, PublicKey=002400000480000094000000060200000024000052534131000400000100010071a8770f460cce31df0feb6f94b328aebd55bffeb5c69504593df097fdd9b29586dbd155419031834411c8919516cc565dee6b813c033676218496edcbe7939c0dd1f919f3d1a228ebe83b05a3bbdbae53ce11bcf4c04a42d8df1a83c2d06cb4ebb0b447e3963f48a1ca968996f3f0db8ab0e840a89d0a5d5a237e2f09189ed3")]
-
 namespace Tmds.DBus.Protocol;
 
-internal ref struct SignatureReader
```
Comment (tmds) [inline src/Tmds.DBus.Protocol/SignatureReader.cs:5]: This change is not related to the multi-segment send.

@affederaffe are you using this? Which member exactly?

File: src/Tmds.DBus.Protocol/MessageStream.cs [hunk 2]
```diff
@@ -94,7 +94,7 @@ private async void ReadMessagesIntoSocket()
                     SequencePosition position = buffer.Start;
                     while (buffer.TryGet(ref position, out ReadOnlyMemory<byte> memory))
                     {
-                        await _socket.SendAsync(buffer.First, handles).ConfigureAwait(false);
+                        await _socket.SendAsync(memory, handles).ConfigureAwait(false);
```
Comment (tmds) [inline src/Tmds.DBus.Protocol/MessageStream.cs:97]: :+1:

File: src/Tmds.DBus.Protocol/SignatureReader.cs (hunk 1 above)
Comment (tmds) [reply src/Tmds.DBus.Protocol/SignatureReader.cs:5]: If you would like to have this type exposed, please make a separate PR, and only make the members you need `public`.

Comment (affederaffe) [reply src/Tmds.DBus.Protocol/SignatureReader.cs]: I think I was using it for the source generator at some point, though I removed it since it creates an explicit dependency on Tmds.DBus.Protocol. Reverted the relevant commit.

**Comment by tmds** [comment]
@affederaffe thanks!
